lib/liquid/context.rb

```diff
@@ -207,6 +207,8 @@ def find_variable(key)
     def lookup_and_evaluate(obj, key)
       if (value = obj[key]).is_a?(Proc) && obj.respond_to?(:[]=)
         obj[key] = (value.arity == 0) ? value.call : value.call(self)
+      elsif strict_variables && obj.respond_to?(:has_key?) && !obj.has_key?(key)
```

Comment:
I think this should be `key?`, not `has_key?` (see https://github.com/Shopify/liquid/pull/593)